LumpNav/LumpNav.py

```diff
@@ -843,101 +856,135 @@ def getViewNode(self, viewName):
+    if (self.viewpointLogic.getViewpointForViewNode(viewNode).isCurrentModeAutoCenter()):
+      self.disableAutoCenterInViewNode(viewNode)
     else:
-      self.viewpointLogic.nodeInstanceDictionary[viewNode].trackViewStop()
-      self.setDisableSliders(True, viewNode)
-    self.updateDisableForButtons()
+      self.enableAutoCenterInViewNode(viewNode)
+    self.updateGUIButtons()
+    
+  def disableAutoCenterInViewNode(self, viewNode):
+    logging.debug("disableAutoCenterInViewNode")
+    if (self.viewpointLogic.getViewpointForViewNode(viewNode).isCurrentModeAutoCenter()):
+      self.viewpointLogic.getViewpointForViewNode(viewNode).autoCenterStop()
+    
+  def enableAutoCenterInViewNode(self, viewNode):
+    logging.debug("enableAutoCenterInViewNode")
+    self.disableViewpointInViewNode(viewNode)
+    heightViewCoordLimits = 0.6;
+    widthViewCoordLimits = 0.9;
+    self.viewpointLogic.getViewpointForViewNode(viewNode).setViewNode(viewNode)
+    self.viewpointLogic.getViewpointForViewNode(viewNode).autoCenterSetSafeXMinimum(-widthViewCoordLimits)
+    self.viewpointLogic.getViewpointForViewNode(viewNode).autoCenterSetSafeXMaximum(widthViewCoordLimits)
+    self.viewpointLogic.getViewpointForViewNode(viewNode).autoCenterSetSafeYMinimum(-heightViewCoordLimits)
+    self.viewpointLogic.getViewpointForViewNode(viewNode).autoCenterSetSafeYMaximum(heightViewCoordLimits)
+    self.viewpointLogic.getViewpointForViewNode(viewNode).autoCenterSetModelNode(self.tumorModel_Needle)
+    self.viewpointLogic.getViewpointForViewNode(viewNode).autoCenterStart()
+    
+  def disableViewpointInViewNode(self,viewNode):
+    logging.debug("disableViewpointInViewNode")
+    self.disableBullseyeInViewNode(viewNode)
+    self.disableAutoCenterInViewNode(viewNode)
       
-  def onRightFollowCameraButtonClicked(self):
-    logging.debug("onRightFollowCameraButtonClicked {0}".format(self.rightFollowCameraButton.isChecked()))
-    viewNode = self.getViewNode('View2')
-    self.viewpointLogic.changeCurrentViewNode(viewNode)
-    if (self.rightFollowCameraButton.isChecked() == True):
-      self.viewpointLogic.nodeInstanceDictionary[viewNode].setViewNode(viewNode)
-      self.viewpointLogic.nodeInstanceDictionary[viewNode].followSetSafeXMinimum(-0.6)
-      self.viewpointLogic.nodeInstanceDictionary[viewNode].followSetSafeXMaximum(0.6)
-      self.viewpointLogic.nodeInstanceDictionary[viewNode].followSetSafeYMinimum(-0.6)
-      self.viewpointLogic.nodeInstanceDictionary[viewNode].followSetSafeYMaximum(0.6)
-      self.viewpointLogic.nodeInstanceDictionary[viewNode].followSetModelNode(self.tumorModel_Needle)
-      self.viewpointLogic.nodeInstanceDictionary[viewNode].followStart()
+  def updateGUIButtons(self):
+    logging.debug("updateGUIButtons")
+    
+    leftViewNode = self.getViewNode('View1')
+    
+    if (self.viewpointLogic.getViewpointForViewNode(leftViewNode).isCurrentModeAutoCenter()):
+      self.leftAutoCenterCameraButton.setChecked(True)

```

Comment:
blockSignals prevents the widget to emit signals when you change it. In this case, probably you want to block the self.leftAutoCenterCameraButton widget to emit any signals when you set its state to checked.

Example:
https://github.com/Slicer/Slicer/blob/master/Libs/MRML/Widgets/qMRMLLinearTransformSlider.cxx#L203..L205